File: Arrays/Algorithms/Sliding_Window/Circular/max_sum_circular_subarray.py
def max_sum_circular_subarray(arr, k):
	n = len(arr)
	max_sum = float('-inf')
	for i in range(n):
		curr_sum = 0
		for j in range(k):
			curr_sum += arr[(i + j) % n]
		max_sum = max(max_sum, curr_sum)
	return max_sum

if __name__ == "__main__":
	n, k = map(int, input().split())
	arr = list(map(int, input().split()))
	print(max_sum_circular_subarray(arr, k))

File: Arrays/Algorithms/Sliding_Window/Circular/max_sum_circular_subarray.cpp
#include <bits/stdc++.h>
using namespace std;

int maxSumCircularSubarray(vector<int>& arr, int k) {
	int n = arr.size();
	int max_sum = INT_MIN;
	for (int i = 0; i < n; i++) {
		int curr_sum = 0;
		for (int j = 0; j < k; j++) {
			curr_sum += arr[(i + j) % n];
		}
		max_sum = max(max_sum, curr_sum);
	}
	return max_sum;
}

int main() {
	int n, k;
	cin >> n >> k;
	vector<int> arr(n);
	for (int i = 0; i < n; i++) cin >> arr[i];
	cout << maxSumCircularSubarray(arr, k) << endl;
	return 0;
}
